src/openloco/windows/stationwnd.cpp

```diff
@@ -1,55 +1,660 @@
+            self->widgets[widx::scrollview].bottom = self->height - 14;
+
+            self->widgets[widx::status_bar].top = self->height - 12;
+            self->widgets[widx::status_bar].bottom = self->height - 3;
+            self->widgets[widx::status_bar].right = self->width - 14;
+
+            self->widgets[widx::station_catchment].right = self->width - 2;
+            self->widgets[widx::station_catchment].left = self->width - 25;
+
+            common::repositionTabs(self);
+        }
+
+        // 0x0048E8DE
+        static void draw(window* self, gfx::drawpixelinfo_t* dpi)
+        {
+            self->draw(dpi);
+            common::drawTabs(self, dpi);
+
+            auto buffer = const_cast<char*>(stringmgr::get_string(string_ids::buffer_1250));
+            buffer = stringmgr::format_string(buffer, string_ids::accepted_cargo_separator);
+
+            auto station = stationmgr::get(self->number);
+            uint8_t cargoTypeCount = 0;
+
+            for (uint32_t cargoId = 0; cargoId < max_cargo_stats; cargoId++)
+            {
+                auto& stats = station->cargo_stats[cargoId];
+
+                if (!stats.is_accepted())
+                    continue;
+
+                *buffer++ = ' ';
+                *buffer++ = control_codes::inline_sprite_str;
+                *(reinterpret_cast<uint32_t*>(buffer)) = objectmgr::get<cargo_object>(cargoId)->unit_inline_sprite;
+                buffer += 4;
+
+                cargoTypeCount++;
+            }
+
+            if (cargoTypeCount == 0)
+            {
+                buffer = stringmgr::format_string(buffer, string_ids::cargo_nothing_accepted);
+            }
+
+            *buffer++ = '\0';
+            auto args = FormatArguments();
+            args.push(string_ids::buffer_1250);
+            
+            const auto& widget = self->widgets[widx::status_bar];
+            const auto x = self->x + widget.left - 1;
+            const auto y = self->y + widget.top - 1;
+            const auto width = widget.width();
+            gfx::draw_string_494BBF(*dpi, x, y, width, colour::black, string_ids::buffer_1250, &args);
+        }
+
+        // 0x0048EB0B
+        static void on_mouse_up(window* self, widget_index widgetIndex)
+        {
+            switch (widgetIndex)
+            {
+                case common::widx::caption:
+                    common::renameStationPrompt(self, widgetIndex);
+                    break;
+
+                case common::widx::close_button:
+                    WindowManager::close(self);
+                    break;
+
+                case common::widx::tab_station:
+                case common::widx::tab_cargo:
+                case common::widx::tab_cargo_ratings:
+                    common::switchTab(self, widgetIndex);
+                    break;
+
+                case widx::station_catchment:
+                    if (self->number == word_112C786)
+                        self->number = 65535;
```

Comment:
This isn't correct. It is passing -1 into regs.ax but it is not setting the self->number to 65535. If you did this it would break a lot of code that assumes the self->number is a valid station number.